services/sideload/service.go

```diff
@@ -258,7 +262,10 @@ type httpSource struct {
 }
 
 func (s *httpSource) UpdateCache() error {
-	s.cache = make(map[string]map[string]interface{})
+	s.fileSource.mu.Lock()
```

Comment:
should we lock for the whole update? seems like if these are happening in parallel it would be bad if we clean the cache, then start updating in one goroutine , then clean in a new goroutine, then finish updating from both goroutines.

Reply:
Good point.